crates/ethrpc/src/block_stream/mod.rs

```diff
@@ -359,6 +364,24 @@ fn update_block_metrics(current_block: u64, new_block: u64) {
     }
 }
 
+/// Records newly observed block number in the metrics.
+fn update_current_block_metrics(block_number: u64) {
+    let metric = &Metrics::instance(observe::metrics::get_storage_registry())
+        .unwrap()
+        .last_block_number;
+
+    match i64::try_from(block_number) {
+        Ok(num) => metric.set(num),
+        Err(err) => {
+            tracing::warn!(
+                ?err,
+                ?block_number,
+                "failed to set last block number metric, the block number is not a valid i64"
+            );
+        }
+    }
```

Comment:
You can avoid this error handling by using a u64 based gauge: `prometheus::core::GenericGauge<prometheus::core::AtomicU64>`